### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MaiCabrera**
- Setup for basic apprunner

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -28,29 +28,42 @@ resource "aws_apprunner_connection" "dev_apprunner" {
...
-  service_name = "example"
+  service_name = "${var.service_name}-${random_uuid.apprunner_service_name_suffix.result}"
 
   source_configuration {
-    image_repository {
-      image_configuration {
-        port = "8000"
+    authentication_configuration {
+      connection_arn = aws_apprunner_connection.example.arn
+    }
```
Comment (GH-CharlesLandau) [inline main.tf:47]: You can get rid of this

File: main.tf [hunk 2]
```diff
@@ -28,29 +28,42 @@ resource "aws_apprunner_connection" "dev_apprunner" {
...
+          port          = "8000"
+          runtime       = "python3"
+          start_command = "python runapp.py"
+        }
+        configuration_source = "API"
+      }
+      repository_url = "https://github.com/example/my-example-python-app"
+      source_code_version {
+        type  = "BRANCH"
+        value = "main"
```
Comment (GH-CharlesLandau) [inline main.tf:61]: I thought we were using `image_repository`?

File: main.tf (hunk 1 above)
Comment (GH-CharlesLandau) [reply main.tf:47]: We talked about this and you're going to go figure out whether we need to put our existing HCL in a code_configuration repo or if we can get away with just pushing commands to an image_configuration

**Review by GH-CharlesLandau — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (MaiCabrera) [reply main.tf:61]: [Apprunner docs](https://github.com/awsdocs/aws-app-runner-developer-guide/blob/main/doc_source/config-file.md)

source code is the only method that will enable configurations. We can use ECR however we would forego all our config we currently have for the vault

File: main.tf (hunk 1 above)
Comment (MaiCabrera) [reply main.tf:47]: code_configuration will be the way to go, link https://github.com/awsdocs/aws-app-runner-developer-guide/blob/main/doc_source/config-file.md

File: main.tf [hunk 3]
```diff
@@ -28,15 +28,15 @@ resource "aws_apprunner_connection" "dev_apprunner" {
 // C. APPRUNNER CUSTOM DOMAIN ASSOC //
 //==================================//
 resource "aws_apprunner_custom_domain_association" "dev_apprunner" {
-  domain_name = "example.com"
-  service_arn = aws_apprunner_service.example.arn
+  domain_name = var.domain_name
+  service_arn = aws_apprunner_service.var.service_arn
```
[the remark this answers is not in the document]
Comment (MaiCabrera) [reply main.tf:32]: thank you for the correction in syntax.

**Review by GH-CharlesLandau — CHANGES_REQUESTED** [review]

File: main.tf [hunk 4]
```diff
@@ -28,29 +28,42 @@ resource "aws_apprunner_connection" "dev_apprunner" {
 // C. APPRUNNER CUSTOM DOMAIN ASSOC //
 //==================================//
 resource "aws_apprunner_custom_domain_association" "dev_apprunner" {
-  domain_name = "example.com"
-  service_arn = aws_apprunner_service.example.arn
+  domain_name = var.domain_name
+  service_arn = aws_apprunner_service.service_arn
```
[the remark this answers is not in the document]
Comment (MaiCabrera) [reply main.tf:32]: updated and changed in new commit.

File: variables.tf [hunk 5]
```diff
@@ -37,61 +36,52 @@ variable "tags" {
...
+
+    authentication_configuration = object ({
+      connection_arn = string
+    })
+
+    code_repository = object ({
+
+    })
+
+  }))
```
[the remark this answers is not in the document]
Comment (MaiCabrera) [reply variables.tf:85]: It didn't and I realized I was majorly messing up this section.

File: main.tf [hunk 6]
```diff
@@ -2,55 +2,86 @@
...
-  max_concurrency = var.max_concurrency
-  max_size        = var.max_size
-  min_size        = var.min_size
+  max_concurrency = "${var.max_concurrency}"
+  max_size        = "${var.max_size}"
+  min_size        = "${var.min_size}"
 
   tags = {
-    Name = var.tags
+    Name = "${var.tags}"
```
Comment (GH-CharlesLandau) [inline main.tf:12]: Incorrect, better to do `tags = var.tags` or something with the `merge` builtin function for all of these values. Have a look at how cloudposse does it

File: main.tf [hunk 7]
```diff
@@ -2,55 +2,86 @@
...
 //===============================//
 resource "aws_apprunner_connection" "dev_apprunner" {
-  connection_name = var.connection_name
-  provider_type   = var.provider_type
-
+  connection_name = "${var.connection_name}"
+  provider_type   = "${var.provider_type}"
   tags = {
-    Name = var.tags
+    Name = "${var.tags}"
```
Comment (GH-CharlesLandau) [inline main.tf:22]: See comment above

File: main.tf [hunk 8]
```diff
@@ -2,55 +2,86 @@
...
 }
 //===============================//
 // B.    APPRUNNER CONNECTION   //
 //===============================//
 resource "aws_apprunner_connection" "dev_apprunner" {
-  connection_name = var.connection_name
-  provider_type   = var.provider_type
-
+  connection_name = "${var.connection_name}"
+  provider_type   = "${var.provider_type}"
```
Comment (GH-CharlesLandau) [inline main.tf:20]: String interpolation again

File: main.tf [hunk 9]
```diff
@@ -2,55 +2,86 @@
...
 }
 
-//===================================//
-// D.    APPRUNNER SERVICE W/IMAGE   //
-//===================================//
 resource "aws_apprunner_service" "dev_apprunner" {
-  service_name = "example"
+  service_name = "${var.service_name}-${random_uuid.apprunner_service_name_suffix.result}"
 
+  auto_scaling_configuration_arn = "${aws_apprunner_auto_scaling_configuration_version.dev_apprunner.arn}"
```
Comment (GH-CharlesLandau) [inline main.tf:43]: String interpolation in this style is deprecated

File: main.tf [hunk 10]
```diff
@@ -2,55 +2,86 @@
...
+
+//ARN IN REQUEST IS THE CONNECTION RESOURCES, IN DOUBT CHECK
```
Comment (GH-CharlesLandau) [inline main.tf:45]: Huh? Please explain

File: main.tf [hunk 11]
```diff
@@ -2,55 +2,86 @@
...
   source_configuration {
-    image_repository {
-      image_configuration {
-        port = "8000"
+    authentication_configuration {
+      connection_arn = "${aws_apprunner_connection.dev_apprunner.arn}"
```
Comment (GH-CharlesLandau) [inline main.tf:48]: String interpolation

File: main.tf [hunk 12]
```diff
@@ -2,55 +2,86 @@
...
+    }
+
+    auto_deployments_enabled {
+      auto_deployment = "${var.auto_deployment}"
```
Comment (GH-CharlesLandau) [inline main.tf:52]: String interpolation. I'm going to stop making this comment but suffice to say please fix this throughout the PR

File: main.tf [hunk 13]
```diff
@@ -2,55 +2,86 @@
...
+    code_repository {
+      code_configuration {
+        code_configuration_values {
+          build_command = "${var.build_command}"
+          port          = "${var.port}"
+          runtime       = "${var.runtime}"
+          start_command = "${var.start_command}"
+        }
+        configuration_source = "${var.configuration_source}"
+      }
```
Comment (GH-CharlesLandau) [inline main.tf:64]: Let's remove `code_configuration_values` and make them always use `REPOSITORY`

File: main.tf [hunk 14]
```diff
@@ -2,55 +2,86 @@
...
+        type  = "${var.type}"
+        value = "${var.value}"
       }
-      image_identifier      = "public.ecr.aws/jg/hello:latest"
-      image_repository_type = "ECR_PUBLIC"
     }
   }
+  
+  encryption_configuration {
+    kms_key = "${var.kms_key}"
```
Comment (GH-CharlesLandau) [inline main.tf:75]: By default, the KMS key should be created as a resource within this module, and then let the user provide a bool variable if they want to use their own, and then if that bool is true then don't create the resource.

File: main.tf [hunk 15]
```diff
@@ -2,55 +2,86 @@
...
+  }
+
+  instance_configuration {
+    cpu = "${var.cpu}"
+    memory = "${var.memory}"
+    instance_role_arn = "${var.instance_role_arn}"
```
Comment (GH-CharlesLandau) [inline main.tf:81]: Nope! This needs to be created

File: main.tf [hunk 16]
```diff
@@ -2,55 +2,86 @@
...
+  }
 
   tags = {
     Name = "example-apprunner-service"
   }
```
Comment (GH-CharlesLandau) [inline main.tf:86]: hardcoded tags no no

File: output.tf [hunk 17]
```diff
@@ -2,113 +2,60 @@
 //      AUTOSCALING OUTPUTS      //
 //===============================//
 output "arn" {
```
Comment (GH-CharlesLandau) [inline output.tf:4]: `arn` is not sufficiently descriptive

File: output.tf [hunk 18]
```diff
@@ -2,113 +2,60 @@
...
-  value = ""
+  value = "${aws_apprunner_auto_scaling_configuration_version.dev_apprunner.arn}"
```
Comment (GH-CharlesLandau) [inline output.tf:5]: Why not return the whole resource instead of just the arn?

File: output.tf [hunk 19]
```diff
@@ -2,113 +2,60 @@
...
+  description = "ARN of the connection for apprunner located in the connection resource output file"
 }
 
 output "auto_scaling_configuration_revision" {
-  value = ""
+  value = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.auto_scaling_configuration_revision
 }
```
Comment (GH-CharlesLandau) [inline output.tf:11]: Why not collapse this and the previous output into one and return the whole resource?

File: output.tf [hunk 20]
```diff
@@ -2,113 +2,60 @@
...
 
-output "latest" {
-  value = ""
-}
 output "status" {
-  value = ""
+  value = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.status
 }
```
Comment (GH-CharlesLandau) [inline output.tf:15]: Maybe we shouldn't do this? Terraform will read state every time

File: output.tf [hunk 21]
```diff
@@ -2,113 +2,60 @@
...
 
-output "" {
-  value = ""
+output "tags_all" {
+  value = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.tags_all
 }
```
Comment (GH-CharlesLandau) [inline output.tf:19]: What is this?

File: output.tf [hunk 22]
```diff
@@ -2,113 +2,60 @@
...
-  value = ""
-}
-
-output "status" {
-  value = ""
-}
-
-output "tags" {
-  value = ""
+  value = "${aws_apprunner_connection.dev_apprunner.arn}"
```
Comment (GH-CharlesLandau) [inline output.tf:25]: Duplicative

File: output.tf [hunk 23]
```diff
@@ -2,113 +2,60 @@
...
-#   value = ""
-# }
-
-# output "" {
-#   value = ""
-# }
-# output "" {
-#   value = ""
-# }
+output "id" {
```
Comment (GH-CharlesLandau) [inline output.tf:32]: Vauge

File: output.tf [hunk 24]
```diff
@@ -2,113 +2,60 @@
...
 
-# output "" {
-#   value = ""
-# }
+output "certificate_validation_records" {
+  value = aws_apprunner_custom_domain_association.dev_apprunner.certificate_validation_records
+}
+output "dns_target" {
+  value = aws_apprunner_custom_domain_association.dev_apprunner.dns_target
+}
```
Comment (GH-CharlesLandau) [inline output.tf:41]: What is this?

File: output.tf [hunk 25]
```diff
@@ -2,113 +2,60 @@
...
+//==============//
+//    SERVICE   //
+//==============//
 
-# output "" {
-#   value = ""
-# }
+output "arn" {
+  value = "${aws_apprunner_service.dev_apprunner.arn}"
+}
```
Comment (GH-CharlesLandau) [inline output.tf:49]: Duplicative

File: output.tf [hunk 26]
```diff
@@ -2,113 +2,60 @@
...
+output "service_url" {
+  value = aws_apprunner_service.dev_apprunner.service_url
+}
 
-# output "" {
-#   value = ""
-# }
+output "status" {
+  value = aws_apprunner_service.dev_apprunner.status
+}
```
Comment (GH-CharlesLandau) [inline output.tf:60]: Duplicative

File: provider.tf [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+  source = "./modules/networking"
+
+  region               = var.region
+  environment          = var.environment
+  vpc_cidr             = var.vpc_cidr
+  public_subnets_cidr  = var.public_subnets_cidr
+  private_subnets_cidr = var.private_subnets_cidr
+  availability_zones   = local.production_availability_zones
+}
+
```
Comment (GH-CharlesLandau) [inline provider.tf:26]: To be removed

File: provider.tf [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+  // encryption configuration 
+  kms_key = ""
+
+  // instance configuration
+  cpu = ""
+  memory = ""
+  instance_role_arn = ""
+
+}
```
Comment (GH-CharlesLandau) [inline provider.tf:88]: If you don't drill values from the `variables.tf` file down into the submodules then they won't propagate into the actual `terraform plan`

File: variables.tf [hunk 29]
```diff
@@ -4,7 +4,6 @@
 variable "autoScalingConfigurationName" {
```
Comment (GH-CharlesLandau) [inline variables.tf:4]: The rest of the PR review suggests so many changes to this that I think you need to reevaluate and I'm not going to review yet if that's ok

**Review by GH-CharlesLandau — CHANGES_REQUESTED** [review]

File: main.tf [hunk 30]
```diff
@@ -2,55 +2,86 @@
 // A. APPRUNNER AUTOSCALING GROUP //
 //===============================//
 resource "aws_apprunner_auto_scaling_configuration_version" "dev_apprunner" {
-  auto_scaling_configuration_name = var.autoScalingConfigurationName
+  auto_scaling_configuration_name = "${var.autoScalingConfigurationName}"
```
[the remark this answers is not in the document]
Comment (MaiCabrera) [reply main.tf:5]: linted

File: main.tf [hunk 31]
```diff
@@ -2,55 +2,86 @@
...
 
-  max_concurrency = var.max_concurrency
-  max_size        = var.max_size
-  min_size        = var.min_size
+  max_concurrency = "${var.max_concurrency}"
+  max_size        = "${var.max_size}"
+  min_size        = "${var.min_size}"
```
[the remark this answers is not in the document]
Comment (MaiCabrera) [reply main.tf:9]: linted

File: main.tf (hunk 7 above)
Comment (MaiCabrera) [reply main.tf:22]: linted

File: output.tf (hunk 25 above)
Comment (MaiCabrera) [reply output.tf:49]: T.T

File: provider.tf (hunk 28 above)
Comment (MaiCabrera) [reply provider.tf]: solved

File: main.tf [hunk 32]
```diff
@@ -2,14 +2,14 @@
...
-  auto_scaling_configuration_name = var.autoScalingConfigurationName
+  auto_scaling_configuration_name = var.auto_scaling_configuration_name
 
   max_concurrency = var.max_concurrency
   max_size        = var.max_size
   min_size        = var.min_size
 
   tags = {
-    Name = var.tags
+    Name = "${var.tags_autoscaling_group}"
```
Comment (GH-CharlesLandau) [inline main.tf:12]: Still interpolating unnecessarily. 

Still needs to align with our convention for tagging: `tags = var.tags` or something with the `merge` builtin function

File: main.tf [hunk 33]
```diff
@@ -18,39 +18,64 @@ resource "aws_apprunner_auto_scaling_configuration_version" "dev_apprunner" {
...
 
+  instance_configuration {
+    cpu               = var.cpu
+    memory            = var.memory
+    instance_role_arn = var.instance_role_arn
+  }
 
+  tags = {
+    Name = "${var.service_tags}"
+  }
```
Comment (GH-CharlesLandau) [inline main.tf:80]: See previous comment about tagging

File: output.tf [hunk 34]
```diff
@@ -1,114 +1,61 @@
 //================================//
 //      AUTOSCALING OUTPUTS      //
 //===============================//
-output "arn" {
-  value = ""
+output "dev_apprunner_autoscaling_arn" {
+  value       = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.arn
+  description = "ARN of the connection for apprunner located in the connection resource output file"
```
Comment (GH-CharlesLandau) [inline output.tf:6]: This seems unclear. What would a user do with this output?

File: output.tf [hunk 35]
```diff
@@ -1,114 +1,61 @@
...
-output "status" {
-  value = ""
+output "dev_apprunner_autoscaling_status" {
+  value = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.status
 }
 
-output "" {
-  value = ""
+output "dev_apprunner_autoscaling_tags_all" {
+  value = aws_apprunner_auto_scaling_configuration_version.dev_apprunner.tags_all
```
Comment (GH-CharlesLandau) [inline output.tf:18]: Why are we doing this as opposed to outputting the whole resource?

File: output.tf [hunk 36]
```diff
@@ -1,114 +1,61 @@
...
-#   value = ""
-# }
-
-# output "" {
-#   value = ""
-# }
-# output "" {
-#   value = ""
-# }
+output "dev_apprunner_custdomain_id" {
```
Comment (GH-CharlesLandau) [inline output.tf:32]: `dev_apprunner_custom_domain_id` We aren't going to run out of ASCII chars 😉

File: output.tf [hunk 37]
```diff
@@ -1,114 +1,61 @@
...
+}
 
-# output "" {
-#   value = ""
-# }
+output "dev_apprunner_custdomain_certificate_validation_records" {
+  value = aws_apprunner_custom_domain_association.dev_apprunner.certificate_validation_records
+}
+output "dev_apprunner_custdomain_dns_target" {
+  value = aws_apprunner_custom_domain_association.dev_apprunner.dns_target
```
Comment (GH-CharlesLandau) [inline output.tf:40]: Descriptions, plus what will users do with this stuff?

File: provider.tf [hunk 38]
```diff
@@ -0,0 +1,60 @@
...
+  max_size                        = 2
+  min_size                        = 1
+  tags_autoscaling_group          = "dev_aws_vault_apprunner"
+
+  // Connection Variables
+  connection_name = "dev_apprunner_aws_vault"
+  provider_type   = "GITHUB"
+
+  // Domain Assoc Variables
+  domain_name = "dev_apprunner_aws"
```
Comment (GH-CharlesLandau) [inline provider.tf:31]: Won't this result in name collision?

File: provider.tf [hunk 39]
```diff
@@ -0,0 +1,60 @@
...
+  auto_scaling_configuration_arn = "${module.apprunner.dev_apprunner_connection_arn}"
+
+  // encryption configuration 
+  kms_key = ""
+
+  // instance configuration
+  cpu               = ""
+  memory            = ""
+  instance_role_arn = ""
+  service_tags      = "dev_aws_vault_apprunner_service_tag"
```
Comment (GH-CharlesLandau) [inline provider.tf:59]: String interpolation again

If the values are hardcoded why are they part of the interface at all? If you're going to hardcode a variable please also have an explanation about why it belongs in `./modules/apprunner/variables.tf` in the first place

File: variables.tf [hunk 40]
```diff
@@ -37,61 +36,52 @@ variable "tags" {
 variable "connection_name" {
   type        = string
   description = "small detail of how var is used"
-  default     = "dev_apprunner"
 }
 
 variable "provider_type" {
   type        = string
   description = "small detail of how var is used"
-  default     = "aws"
```
[the remark this answers is not in the document]
Comment (GH-CharlesLandau) [reply variables.tf:46]: There are still lots of unused vars, see previous comment about hardcoding

File: variables.tf [hunk 41]
```diff
@@ -37,61 +35,88 @@ variable "tags" {
...
 }
-variable "tags" {
+variable "connection_tags" {
   type        = string
   description = "small detail of how var is used"
```
Comment (GH-CharlesLandau) [inline variables.tf:46]: Is this a placeholder string? Pls remove

File: variables.tf [hunk 42]
```diff
@@ -37,61 +35,88 @@ variable "tags" {
...
+variable "memory" {
+  type = number
+}
+
+variable "instance_role_arn" {
+  type = string
+}
+
+variable "service_tags" {
+  type = string
```
Comment (GH-CharlesLandau) [inline variables.tf:121]: No descriptions

**Review by GH-CharlesLandau — CHANGES_REQUESTED** [review]

File: main.tf (hunk 32 above)
Comment (MaiCabrera) [reply main.tf:12]: oversight on my end to not run the tf fmt.

File: main.tf (hunk 33 above)
Comment (MaiCabrera) [reply main.tf:80]: oversight on my end to not run the tf fmt.

File: variables.tf (hunk 41 above)
Comment (MaiCabrera) [reply variables.tf:46]: its not a placeholder, its referenced by a resource. I'll add more contexts in the description

File: provider.tf (hunk 39 above)
Comment (MaiCabrera) [reply provider.tf:59]: Should not be hardcoded, I've updated so that the interpolation string is removed.

File: output.tf (hunk 35 above)
Comment (MaiCabrera) [reply output.tf]: commented this out as its not really leveraged or used anywhere.

File: output.tf (hunk 34 above)
Comment (MaiCabrera) [reply output.tf:6]: don't believe its used anywhere in the resources.  Should outputs not utilized be nixed?

File: output.tf (hunk 36 above)
Comment (MaiCabrera) [reply output.tf:32]: thought we get charged a penny for each new char

File: provider.tf (hunk 38 above)
Comment (MaiCabrera) [reply provider.tf:31]: will change upon getting all elements ready to work

File: main.tf [hunk 43]
```diff
@@ -1,56 +1,94 @@
...
+  autoscaling_tags = {
+    name = var.autoscaling_tags
+  }
+  connection_tags = {
+    name = var.connection_tags
+  }
+  service_tags = {
+    name = var.service_tags
+  }
+}
```
Comment (GH-CharlesLandau) [inline main.tf:20]: Cool

File: main.tf [hunk 44]
```diff
@@ -1,56 +1,94 @@
...
+  auto_scaling_configuration_name = var.auto_scaling_configuration_name
 
   max_concurrency = var.max_concurrency
   max_size        = var.max_size
   min_size        = var.min_size
 
-  tags = {
-    Name = var.tags
-  }
+  tags = merge(local.apprunner_tags, local.autoscaling_tags)
```
Comment (GH-CharlesLandau) [inline main.tf:31]: What about `var.tags`

File: main.tf [hunk 45]
```diff
@@ -1,56 +1,94 @@
...
+  tags            = merge(local.apprunner_tags, local.connection_tags)
 }
 
 //===================================//
 // C. APPRUNNER CUSTOM DOMAIN ASSOC //
 //==================================//
 resource "aws_apprunner_custom_domain_association" "dev_apprunner" {
-  domain_name = "example.com"
-  service_arn = aws_apprunner_service.example.arn
+  domain_name = var.domain_name
```
Comment (GH-CharlesLandau) [inline main.tf:46]: Random suffix?

File: main.tf [hunk 46]
```diff
@@ -1,56 +1,94 @@
...
+  service_arn = var.service_arn
```
Comment (GH-CharlesLandau) [inline main.tf:47]: Shouldn't this reference `aws_apprunner_service.dev_apprunner.arn`

File: main.tf [hunk 47]
```diff
@@ -1,56 +1,94 @@
...
 
+  //ARN IN REQUEST IS THE CONNECTION RESOURCES, IN DOUBT CHECK
   source_configuration {
-    image_repository {
-      image_configuration {
-        port = "8000"
+    auto_deployments_enabled = var.auto_deployments_enabled
+
+    authentication_configuration {
+      connection_arn = var.connection_arn
```
Comment (GH-CharlesLandau) [inline main.tf:67]: Shouldn't this reference `aws_apprunner_connection.dev_apprunner.arn`

File: main.tf [hunk 48]
```diff
@@ -1,56 +1,94 @@
...
   }
-}
 
+  instance_configuration {
+    cpu               = var.cpu
+    memory            = var.memory
+    instance_role_arn = var.instance_role_arn
+  }
 
+  tags = merge(local.apprunner_tags, local.service_tags)
```
Comment (GH-CharlesLandau) [inline main.tf:93]: What about var.tags

File: main.tf [hunk 49]
```diff
@@ -1,56 +1,94 @@
...
-    Name = "example-apprunner-service"
+  encryption_configuration {
+    kms_key = var.kms_key
   }
-}
 
+  instance_configuration {
+    cpu               = var.cpu
+    memory            = var.memory
+    instance_role_arn = var.instance_role_arn
```
Comment (GH-CharlesLandau) [inline main.tf:90]: Per our convo about scoped modules, the instance role should be created in here, scoped to this module, and then outputted so the end user can attach more permissions if desired

File: variables.tf [hunk 50]
```diff
@@ -1,13 +1,30 @@
...
+variable "project" {
+  type        = string
+  description = "name of project you are creating"
+  default     = "ccl-dev-apprunner"
+}
+variable "owner" {
+  type        = string
+  description = "name of project you are creating"
+  default     = "ccl, developer_maria cabrera"
+}
```
Comment (GH-CharlesLandau) [inline variables.tf:18]: This breaks our usual convention of using the CCL aws project module as an input to define these things

File: variables.tf [hunk 51]
```diff
@@ -37,61 +54,101 @@ variable "tags" {
...
-  default     = 
+
 }
-variable "" {
-  type        = list
+variable "service_arn" {
+  type        = string
   description = "small detail of how var is used"
-  default     = 
 }
```
Comment (GH-CharlesLandau) [inline variables.tf:80]: Incorrect, the service arn is generated within the module

File: variables.tf [hunk 52]
```diff
@@ -37,61 +54,101 @@ variable "tags" {
...
+variable "repository_url" {
+  type        = string
+  description = "The location of the repository that contains the source code."
+}
+
+variable "type" {
+  type        = string
+  description = "The type of version identifier. For a git-based repository, branches represent versions. Valid values: BRANCH"
+}
+
```
Comment (GH-CharlesLandau) [inline variables.tf:117]: Why is this a required variable? Why is it a variable at all?

File: variables.tf [hunk 53]
```diff
@@ -37,61 +54,101 @@ variable "tags" {
...
+variable "value" {
```
Comment (GH-CharlesLandau) [inline variables.tf:118]: Not descriptive

**Review by GH-CharlesLandau — CHANGES_REQUESTED** [review]

File: main.tf (hunk 45 above)
Comment (MaiCabrera) [reply main.tf:46]: random suffix added

File: main.tf (hunk 47 above)
Comment (MaiCabrera) [reply main.tf:67]: updated connection_arn to ingest attribute + removal of unnecessary var

File: variables.tf (hunk 50 above)
Comment (MaiCabrera) [reply variables.tf:18]: fixed and removed unnecessary var

File: variables.tf (hunk 53 above)
Comment (MaiCabrera) [reply variables.tf:118]: removed as type and value been hardcoded

File: variables.tf (hunk 51 above)
Comment (MaiCabrera) [reply variables.tf:80]: fixed

**Pull request closed without merging**